presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -90,14 +90,35 @@ private static boolean shouldRewriteTableLayout(TableScanNode source)
         @Override
         public Result apply(FilterNode filterNode, Captures captures, Context context)
         {
+            TableScanNode tableScan = captures.get(TABLE_SCAN);
+
             TableLayoutRewriter tableLayoutRewriter = new TableLayoutRewriter(metadata, context.getSession(), context.getSymbolAllocator(), context.getIdAllocator());
-            PlanNode rewrittenTableScan = tableLayoutRewriter.planTableScan(captures.get(TABLE_SCAN), filterNode.getPredicate());
+            PlanNode rewritten = tableLayoutRewriter.planTableScan(captures.get(TABLE_SCAN), filterNode.getPredicate());
+
+            if (arePlansSame(filterNode, tableScan, rewritten)) {
+                return Result.empty();
+            }
+
+            return Result.ofPlanNode(rewritten);
+        }
+
+        private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNode rewritten)
+        {
+            if (!(rewritten instanceof FilterNode)) {
+                return false;
+            }
+
+            FilterNode rewrittenFilter = (FilterNode) rewritten;
+            if (!Objects.equals(filter.getPredicate(), rewrittenFilter.getPredicate())) {
```

Comment:
I know I'm late to the party here, but can you explain when this works vs. when ExpressionEquivalence is needed?  Trying to understand the difference between this and what was in https://github.com/prestodb/presto/pull/8735

Reply:
See https://github.com/prestodb/presto/pull/9705#discussion_r160608829:
> .. Of course what you [initially] wrote is better, but it looks like this method needs to be as elaborate as required and is still might not be super future-proof ..